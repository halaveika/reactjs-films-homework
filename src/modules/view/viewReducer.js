import { SET_GENRE, SET_LIST_FLEX_DIRECTION } from '../../constants';

export const initialState = {
  isRow: true,
  genre: '',
};

export const viewReducer = (state = initialState, action) => {
  switch (action.type) {
    case SET_LIST_FLEX_DIRECTION:
      return {...state, isRow: action.payload};
    case SET_GENRE:
      return { ...state, genre: action.payload };
    default:
      return state;
  }
};
