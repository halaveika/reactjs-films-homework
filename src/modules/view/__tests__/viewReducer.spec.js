import { viewReducer, initialState } from '../viewReducer';

describe('viewt reducer', () => {
  it('should return the initial state', () => {
    expect(viewReducer(undefined, {})).toMatchSnapshot();
  });

  it('should handle SET_LIST_FLEX_DIRECTION', () => {
    const action = {
      type: 'SET_LIST_FLEX_DIRECTION',
      payload: true,
    };
    expect(
      viewReducer(initialState,
        action),
    ).toMatchSnapshot();
  });

  it('should handle SET_GENRE', () => {
    const action = {
      type: 'SET_GENRE',
      payload: 'Drama',
    };
    expect(
      viewReducer(initialState,
        action),
    ).toMatchSnapshot();
  });
});
