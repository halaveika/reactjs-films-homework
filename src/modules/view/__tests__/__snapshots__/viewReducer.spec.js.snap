// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`viewt reducer should handle SET_GENRE 1`] = `
Object {
  "genre": "Drama",
  "isRow": true,
}
`;

exports[`viewt reducer should handle SET_LIST_FLEX_DIRECTION 1`] = `
Object {
  "genre": "",
  "isRow": true,
}
`;

exports[`viewt reducer should return the initial state 1`] = `
Object {
  "genre": "",
  "isRow": true,
}
`;
